torch/optim/adam.py

```diff
@@ -537,15 +544,29 @@ def _multi_tensor_adam(
                 )
 
         # Decay the first and second moment running average coefficient
-        torch._foreach_lerp_(device_exp_avgs, device_grads, 1 - beta1)
+        # Use device beta1 if beta1 is a tensor to ensure all
+        # tensors are on the same device
+        torch._foreach_lerp_(device_exp_avgs, device_grads, 1 - device_beta1)
 
         torch._foreach_mul_(device_exp_avg_sqs, beta2)
+
+        # Due to the strictness of the _foreach_addcmul API, we can't have a single
```

Comment:
I wonder if we can add an overload. `scaled_device_grads` being out-of-place is going to use even more intermediates.

Reply:
If we compile these intermediates go away, but agreed that for eager this is a problem and another overload could be a good solution.

Reply:
Well actually now that I think of it perhaps users will only wrap the beta if they're compiling anyway, there isn't really a use case for tensor betas in eager.